x-pack/plugins/security/server/user_profile/index.ts

```diff
@@ -16,3 +21,24 @@ export type {
   UserProfileGetCurrentParams,
 } from './user_profile_service';
 export type { UserProfileGrant } from './user_profile_grant';
+
+interface SetupUiSettingsServiceParams {
+  uiSettingServiceStart: CoreStart['uiSettings'];
+  userSettingsServiceStart: UserSettingServiceStart;
+}
+
+/**
+ * Used to set the provider that contains the factory that creates UserProfileClients
+ *
+ * @param userSettingsServiceStart A service that provides functions to get the current user's profile, that will be
+ * exposed by the {@link UserProfileSettingsClient}
+ * @param uiSettingServiceStart The service that will the input provider to create instances of {@link UserProfileSettingsClient}
+ */
+export function setupUserProfilesClientOnUiSettingsServiceStart({
+  userSettingsServiceStart,
+  uiSettingServiceStart,
+}: SetupUiSettingsServiceParams) {
+  uiSettingServiceStart.setUserProfileSettingsClientFactoryProvider(() => () => {
```

Comment:
I was following the `ClientFactoryProvider` patter I saw utilized with the SO client, but this could probably be simplified.